p2p/src/peer_manager/mod.rs

```diff
@@ -123,20 +133,99 @@ where
         version == self.chain_config.version()
     }
 
+    fn is_peer_address_valid(&self, address: &PeerAddress) -> bool {
+        // The IP must be globally routable
+        match &address {
+            PeerAddress::Ip4(socket) => {
+                std::net::Ipv4Addr::from(socket.ip).is_global_unicast_ip()
+                    || *self.p2p_config.discover_private_ips
+            }
+            PeerAddress::Ip6(socket) => {
+                std::net::Ipv6Addr::from(socket.ip).is_global_unicast_ip()
+                    || *self.p2p_config.discover_private_ips
+            }
+        }
+    }
+
+    /// Discover public addresses for this node after new outbound connection is made
```

Comment:
connections are made

Reply:
Sorry, but are you sure? It's called every time when new connection is established. So I think "connections are made" will be wrong here. I used "addresses" because node may be listening on multiple ports so multiple addresses will be advertised.
Actually it just occurred to me that we should not try to change the port if node is already listening on a global IP (e.g. if a user has specified it from the command line). I will update this next time.

Reply:
Fixed, thanks :smile: